internal/xds/bootstrap/tlscreds/bundle_test.go

```diff
@@ -96,3 +102,79 @@ func (s) TestFailingProvider(t *testing.T) {
 		t.Errorf("EmptyCall() got err: %s, want err to contain: %s", err, wantErr)
 	}
 }
+
+func (s) TestSPIFFEVerifyFuncMismatchedCert(t *testing.T) {
```

Comment:
Can/Should these tests be e2e style as well? That way, we can actually verify that the error returned to the user (as part of the RPC) has enough details in it for the user to take meaningful action.